src/librustc/front/map/mod.rs

```diff
@@ -571,6 +641,11 @@ impl<'ast> Map<'ast> {
     fn with_path_next<T, F>(&self, id: NodeId, next: LinkedPath, f: F) -> T where
         F: FnOnce(PathElems) -> T,
     {
+        // This function reveals the name of the item and hence is a
+        // kind of read. This is inefficient, since it walks ancestors
+        // and we are walking them anyhow, but whatever.
+        self.read(id);
```

Comment:
@michaelwoerister I seem to recall going through a very similar line of thinking when I wrote that call in the first place; maybe I will expand the comment a bit.